platform/security/core/security-core-impl/src/main/resources/OSGI-INF/metatype/metatype.xml

```diff
@@ -0,0 +1,32 @@
+<!--
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+-->
+<metatype:MetaData xmlns:metatype="http://www.osgi.org/xmlns/metatype/v1.0.0">
+
+    <OCD description="Security Username Mapping"
+         name="Security Username Mapping"
+         id="ddf.security.service">
+
+        <AD name="Security Username Mapping" id="usernameAttributeList"
+            required="true"
+            type="String" cardinality="-100"
+            default="http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier, uid"
+            description="List of attributes that are considered for replacing the username of the logged in user.
+            If any of these attributes match any of the attributes within the SecurityAssertion, the value of the first matching attribute will be used as the username.
+            (Does not apply when NameIDFormat is of the following: X509, persistent, kerberos or unspecified, and the username is not empty).-"/>
+    </OCD>
+
+    <Designate pid="ddf.security.service">
```

Comment:
This pid seems a little too generic.  Maybe ddf.security.service.SecurityManager.